lib/core/src/Cardano/Wallet.hs

```diff
@@ -2310,6 +2413,11 @@ data ErrStartTimeLaterThanEndTime = ErrStartTimeLaterThanEndTime
     , errEndTime :: UTCTime
     } deriving (Show, Eq)
 
+data ErrCreateMigrationPlan
+    = ErrCreateMigrationPlanEmpty
```

Comment:
Couldn't it just return and empty list of transactions? That seems fairly obvious and easy to handle for users.

Reply:
You're right, it could.

However, the `NothingToMigrate` condition was already a specific error in the old `migrateWallet` API. This PR just reuses the same approach for `createMigrationPlan`.

It covers two cases:
- The wallet is genuinely empty (therefore there really is nothing available to migrate).
- The wallet isn't empty, but the migration algorithm wasn't able to pay for any of the UTxO entries to be migrated, and therefore the plan is empty.

For the second case, I think it's useful to have a specific error here, as we can provide a useful hint to the user:

> I wasn't able to construct a migration plan. This could be because your wallet is empty, or it could be because the amount of ada in your wallet is insufficient to pay for any of funds to be migrated. Try adding some ada to your wallet before trying again."

For comparison, he **_original_** message was:

>  I can't migrate the wallet with the given ID, because it's either empty or full of small coins which wouldn't be worth migrating."

Reply:
Arguably it might want to be worth including the case-specific `balanceLeftover` in the 403 case. 

Could we have a err403 *but* returning `ApiWalletMigrationPlan` with `selections = []`? It couldn't include an error message though, and perhaps a bit weird, so probably not.

Reply:
> Could we have a err403 but returning ApiWalletMigrationPlan with selections = []? It couldn't include an error message though, and perhaps a bit weird, so probably not.

In that specific case, the `balanceLeftover` would be precisely the same as the wallet's balance. So perhaps it wouldn't be providing any more info than the user already has. (Presumably, they already know their balance, or can easily check it.)